lib/email/bounces.js

```diff
@@ -52,56 +45,65 @@ module.exports = function (log, error) {
+      // EITHER bounce OR complaint here.
+      if (message.bounce) {
         recipients = message.bounce.bouncedRecipients
-      }
-      else if (message.complaint && message.complaint.complaintFeedbackType === 'abuse') {
+      } else if (message.complaint) {
         recipients = message.complaint.complainedRecipients
       }
 
       // SES can now send custom headers if enabled on topic.
       // Headers are stored as an array of name/value pairs.
       // Log the `X-Template-Name` header to help track the email template that bounced.
       // Ref: http://docs.aws.amazon.com/ses/latest/DeveloperGuide/notification-contents.html
-      var templateName = utils.getHeaderValue('X-Template-Name', message)
+      const templateName = utils.getHeaderValue('X-Template-Name', message)
+      const language = utils.getHeaderValue('Content-Language', message)
 
       return P.each(recipients, function (recipient) {
-
-        var email = recipient.emailAddress
-        var logData = {
+        const email = eaddrs.parseOneAddress(recipient.emailAddress).address
+        const logData = {
           op: 'handleBounce',
           action: recipient.action,
           email: email,
           bounce: !! message.bounce,
           diagnosticCode: recipient.diagnosticCode,
           status: recipient.status
         }
+        const bounce = {
+          email: email
+        }
 
         // Template name corresponds directly with the email template that was used
         if (templateName) {
           logData.template = templateName
         }
 
+        if (language) {
+          logData.lang = language
+        }
+
         // Log the type of bounce that occurred
         // Ref: http://docs.aws.amazon.com/ses/latest/DeveloperGuide/notification-contents.html#bounce-types
         if (message.bounce && message.bounce.bounceType) {
-          logData.bounceType = message.bounce.bounceType
+          bounce.bounceType = logData.bounceType = message.bounce.bounceType
 
-          if (message.bounce && message.bounce.bounceSubType) {
-            logData.bounceSubType = message.bounce.bounceSubType
+          if (message.bounce.bounceSubType) {
+            bounce.bounceSubType = logData.bounceSubType = message.bounce.bounceSubType
           }
-        }
-
-        // Log the type of complaint and userAgent reported
-        if (message.complaint) {
+        } else if (message.complaint) {
+          // Log the type of complaint and userAgent reported
           logData.complaint = !! message.complaint
+          bounce.bounceType = 'Complaint'
+
 
           if (message.complaint.userAgent) {
             logData.complaintUserAgent = message.complaint.userAgent
           }
 
           if (message.complaint.complaintFeedbackType) {
-            logData.complaintFeedbackType = message.complaint.complaintFeedbackType
+            bounce.bounceSubType = logData.complaintFeedbackType = message.complaint.complaintFeedbackType
           }
         }
```

Comment:
Any code executing in an `else` clause down here would require a bug in V8. There is already an `if` branch above based on `bounce` or `complaint` which is used to grab the recipients, so there should be no possible way this code would execute otherwise.

Note that the message may alternatively contain a `delivery` object, but if so, we don't currently have any actions to take, and so this event handler just ends with deleting the message. According [SES docs](http://docs.aws.amazon.com/ses/latest/DeveloperGuide/notification-contents.html), the `bounce`, `complaint`, and `delivery` objects are mutually exclusive. A message will only contain one of them.